### ISSUE TITLE
[Discord] Add reference detector

### ISSUE THREAD

**Issue by blueridger** [sourcecred/sourcecred#2736]
Currently, the Discord plugin does not have a reference detector (see [this code TODO](https://github.com/sourcecred/sourcecred/blob/6fa1e2efa4e4134bcbb6cfc8e155a99337033528/src/plugins/discord/plugin.js#L87)). The biggest problem this presents is that discord users cannot be referenced in the initiatives plugin (which is a current need by @youngkidwarrior ), though there are many other less-urgent issues with this.

**Comment by youngkidwarrior** [comment]
@blueridger one note on this is I believe it will only work with specific discord messages, as SourceCred sits now. Discord doesn't have links fo profiles.

**Comment by blueridger** [comment]
That's not true. https://discordapp.com/users/<userId> is a valid URL that takes you to someone's profile in the browser app.

**Comment by youngkidwarrior** [comment]
@blueridger ah lovely! I wasn't sure if that would be sufficient for a profile.
